editor-components/feature/feature9/feature9.tsx

```diff
@@ -1,192 +1,289 @@
-import * as React from "react";
+import React, { useState,useEffect, useRef } from "react";
 import ComposerLink from "../../../../custom-hooks/composer-base-components/Link/link";
 import { BaseFeature } from "../../EditorComponent";
 import styles from "./feature9.module.scss";
+import { colors } from "@mui/material";
+import { object, string } from "yup";
 
-type Horizontals = {
+
+type VerticalItem = {
+  id: string;
   title: string;
   description: string;
-  buttonText: string;
   image: string;
-  link: string;
 };
+
+type Verticals = {
+  id: string;
+  horizentals: VerticalItem[];
+};
+
+type ObserverCardProps = {
+  vertical_item: VerticalItem;
+  onIntersect: (id: string,isVisible:boolean) => void;
+};
+
+const ObserverCard: React.FC<ObserverCardProps> = ({ vertical_item, onIntersect }) => {
```

Comment:
we do not use functional components. remove this.

Reply:
I need to use IntersectionObserver to build my structure